src/Psalm/Internal/Type/TypeCombiner.php

```diff
@@ -334,6 +335,11 @@ public static function combine(
         }
 
         if ($combination->named_object_types !== null) {
+            foreach ($combination->value_types as $key => $atomic_type) {
+                if ($atomic_type instanceof TEnumCase && isset($combination->named_object_types[$atomic_type->value])) {
```

Comment:
That's not great :( Maybe we should have a different syntax for enum cases?

But thinking about it, it may not be too bad. The type from docblock will be collapsed at some point anyway, it may not be a problem

Reply:
> Maybe we should have a different syntax for enum cases?

Well, why? Enum cases _are_ constants, so it makes sense to use the same syntax.

> The type from docblock will be collapsed at some point anyway

Doesn't seem to be happening, even in your branch: https://psalm.dev/r/1698218e19

Reply:
> why

Well, exactly because of what you're describing. An Enum and a property of an enum should be collapsed but this is not the case for constants

Reply:
That doesn't mean we should introduce a new syntax for cases.

Reply:
Not sure what we can do then. I'll merge this for now, it's at least a good first step